### ISSUE TITLE
`Image::to_pixels` results in "bytes per row does not respect `COPY_BYTES_PER_ROW_ALIGNMENT`" error

### ISSUE THREAD

**Issue by 0----0** [ggez/ggez#1135]
Attempting to call `Image::to_pixels` results in this error:

```
thread 'main' panicked at 'wgpu error: Validation Error

Caused by:
    In CommandEncoder::copy_texture_to_buffer
    Copy error
    bytes per row does not respect `COPY_BYTES_PER_ROW_ALIGNMENT`
```

**To Reproduce**
```rust
use std::{env, path};

use ggez::{ContextBuilder, GameResult};

fn main() -> GameResult<()> {
    let resource_dir = if let Ok(manifest_dir) = env::var("CARGO_MANIFEST_DIR") {
        let mut path = path::PathBuf::from(manifest_dir);
        path.push("resources");
        path
    } else {
        path::PathBuf::from("./resources")
    };

    let mut builder = ContextBuilder::new("test", "brakts").add_resource_path(resource_dir);
    let (mut ctx, event_loop) = builder.build()?;
    let image = ggez::graphics::Image::from_path(&mut ctx, "/palette0.png")?;
    let pixels = image.to_pixels(&mut ctx)?;
    println!("Hello, world!");
    Ok(())
}
```
Example image:
![palette0](https://user-images.githubusercontent.com/9070490/200129735-930160cc-fb42-4859-9541-a0018a19016a.png)

**Hardware and Software:**
 - ggez version: 0.8.1
 - OS: Windows 11
 - Graphics card: Radeon 6900 XT

**Comment by PSteinhaus** [comment]
So this seems to be caused by https://docs.rs/wgpu-types/0.14.1/wgpu_types/constant.COPY_BYTES_PER_ROW_ALIGNMENT.html

If we read the docs [here](https://docs.rs/wgpu-types/0.14.1/wgpu_types/struct.ImageDataLayout.html#structfield.bytes_per_row) they say:
> Bytes per “row” in an image.
>
>A row is one row of pixels or of compressed blocks in the x direction.
>
>This value is required if there are multiple rows (i.e. height or depth is more than one pixel or pixel block for compressed textures)
>
>Must be a multiple of 256 for [CommandEncoder::copy_buffer_to_texture](https://docs.rs/wgpu-types/0.14.1/wgpu/struct.CommandEncoder.html#method.copy_buffer_to_texture) and [CommandEncoder::copy_texture_to_buffer](https://docs.rs/wgpu-types/0.14.1/wgpu/struct.CommandEncoder.html#method.copy_texture_to_buffer). You must manually pad the image such that this is a multiple of 256. It will not affect the image data.
>
>[Queue::write_texture](https://docs.rs/wgpu-types/0.14.1/wgpu/struct.Queue.html#method.write_texture) does not have this requirement.
>
>Must be a multiple of the texture block size. For non-compressed textures, this is 1.

So... we manually have to pad images before copying them to a host buffer?... Ok...

**Comment by Vixeliz** [comment]
Closing as it was fixed by #1210
